SimplenoteTests/NSStringSimplenoteTests.swift

```diff
@@ -28,17 +38,29 @@ class NSStringSimplenoteTests: XCTestCase {
         let sampleA = NSString(stringLiteral: "\u{0073}\u{0323}\u{0307}")
         let sampleB = NSString(stringLiteral: "\u{0073}\u{0307}\u{0323}")
         let sampleC = NSString(stringLiteral: "\u{1E69}")
+        let sampleD = NSString(stringLiteral: "\u{0065}\u{0301}")
+        let sampleE = NSString(stringLiteral: "\u{00E9}")
```

Comment:
I think it's better to separate this test in two, because it is testing 2 sets of input data which are not related to each other. 
We can extract a function that would accept an array of strings and test them according to existing logic: check that strings in array are not equal and check that hashes are equal. After that we can call this function with different sets of data from our test.

Example pseudo code:
```
func testNonEqualStringsHaveTheSameHash(_ samples: [String]) { ... }
func test...() {
    testNonEqualStringsHaveTheSameHash(["\u{0073}\u{0323}\u{0307}", "\u{0073}\u{0307}\u{0323}", ""\u{1E69}""])
    testNonEqualStringsHaveTheSameHash(["\u{0065}\u{0301}", "\u{00E9}"])
}
```

Reply:
That is a good idea for separating the different values that should test the same.  I updated the tests, can you check it out again when you get a chance and see what you think?

I'm not sure it is the clearest on what it is testing, if you have any suggestions for improving the readability that would be great!

Reply:
Looks good! But we can do better! :-)

First we can make `func testNonEqualStringsCreateSameHash(_ samples: [NSString])` private, because it's our internal helper method.

To make it more readable, we can combine samples in array like this:
```
let differentStringsOneHashSamples = [
    ["\u{0073}\u{0323}\u{0307}", "\u{0073}\u{0307}\u{0323}", "\u{1E69}"], // ṩ
    ["\u{0065}\u{0301}", "\u{00E9}"] // é
]
```

And in the test we can walk through our samples and call helper method `testNonEqualStringsCreateSameHash`. I think we can also rename the original test `testByEncodingTagAsHashAllowsUsToProperlyCompareStringsThatEvaluateAsNotEqualOtherwise` to `testNonEqualStringsCreateSameHash` but without parameters.

In `differentStringsOneHashSamples` I used swift strings instead of NSString, I think we don't really need to stick to NSString for most of the test, so it's fine to use Swift strings. The only "issue" is that comparing our samples using swift strings would return `true` as it seems that swift strings compare "characters" and not  underlying representation of the string. To solve this we can temporarily cast our strings to NSString: `XCTAssertFalse((samples[count] as NSString).equal(to: samples[testCount]))`

Another thing is that in original test we were comparing every sample to every sample, but I don't think it's needed. I think we can compare the first sample to all other samples and get the same coverage. If A == B and A == C, then B == C. This way we will need to walk the samples array only once.

p.s. you can use for loop instead of while + separate var: `for i in 0..<samples.count`